theano/compile/builders.py

```diff
@@ -168,30 +159,154 @@ def infer_shape(self, node, shapes):
+        - list : each function must return a single Variable. The order
+            of the list must corresponds to inputs
+
+    TODO:
+        - examples for a multi-layer mlp. where?
+        - __hash__, __eq__ otherwise won't merge, try
+          gof.opt.is_same_graph_with_merge(op1.local_outputs, op2,
+          local_outputs)
+        - c_code() to remove the double overhead?
+        - grad() make it support DisconnectedType and the new interface
+        - check how it works with updates.
+        - add test with constant as input or inside the inner graph.
+        - Add support for the GPU? Probably just need an opt to remove transfer
+        - Add support to pickle this Op.
+        - Add support/test with random generator
+        - Recursion detection to prevent Op "forkbomb", either set depth
+          limit or manually check them.
+
+    Notes
+    -----
+    - We support shared variables in the inner graph. This is automatic
+      and invisible to the user. They can be as input to the node or in
+      the inner graph.
+    - We support unused inputs. This is needed for the grad.
+    - `inline=True` will cause better runtime optimization at the cost
+      of compilation time. Like "inline" keyword in C, this is merely a
+      suggestion to compiler which is not guaranteed. Currently only
+      works with "fast_compile" or "fast_run" mode.
+
+    Examples
+    --------
+
+    Example 1:
+
+    .. code-block:: python
+
+        from theano import function, op_from_graph, tensor
+        x, y, z = tensor.scalars('xyz')
+        e = x + y * z
+        op = op_from_graph([x, y, z], [e])
+        # op behaves like a normal theano op
+        e2 = op(x, y, z) + op(z, y, x)
+        fn = function([x, y, z], [e2])
+
+    Example 2 with shared variable:
+
+    .. code-block:: python
+
+        import numpy
+        import theano
+        from theano import config, function, op_from_graph, tensor
+        x, y, z = tensor.scalars('xyz')
+        s = theano.shared(numpy.random.rand(2, 2).astype(config.floatX))
+        e = x + y * z + s
+        op = op_from_graph([x, y, z], [e])
+        # op behaves like a normal theano op
+        e2 = op(x, y, z) + op(z, y, x)
+        fn = function([x, y, z], [e2])
+
+    Example 3 override gradient
+
+    .. code-block:: python
+
+        from thenao import funciton, op_from_graph, tensor, grad
+        x, y, z = tensor.scalars('xyz')
+        e = x + y * z
+        def rescale_dy(inps, grads):
+            x, y, z = inps
+            g = grads
+            return z*2
+        op = op_from_graph(
+            [x, y, z], [e], grad_overrides=[None, rescale_dy, None])
+        e2 = op(x, y, z)
+        dx, dy, dz = grad(e2, [x, y, z])
+        fn = function([x, y, z], [dx, dy, dz])
+        # the graident wrt y is now doubled
+        fn(2., 3., 4.) # [1., 8., 3.]
+
+    """
+    return OpFromGraph(
+        inputs, outputs, inline=inline, grad_overrides=grad_overrides, **kwargs)
```

Comment:
Since `op_from_graph` directly returns `OpFromGraph`, it may be better to keep the documentation in `OpFromGraph` and set `op_from_graph = OpFromGraph` as an alias in the end, if anything.